client/src/main/kotlin/io/spine/examples/pingh/client/LoginFlow.kt

```diff
@@ -0,0 +1,285 @@
+ *
+ * @param client enables interaction with the Pingh server.
+ * @param session the information about the current user session.
+ */
+public class LoginFlow internal constructor(
+    private val client: DesktopClient,
+    private val session: MutableStateFlow<UserSession?>
+) {
+    /**
+     * The current state of the login process.
+     */
+    public val state: MutableStateFlow<LoginState> = MutableStateFlow(LoginState.USERNAME_ENTERING)
+
+    /**
+     * Stores the event received after the user enters their name.
+     *
+     * This is required to initialize the verification state flow.
+     */
+    private var userCodeReceived: UserCodeReceived? = null
+
+    /**
+     * Initiates the username entering state of the login process.
+     *
+     * @throws IllegalStateException if the state of the login flow is not `USERNAME_ENTERING`.
+     */
+    public fun startUsernameEnteringFlow(): UsernameEnteringFlow {
+        check(state.value == LoginState.USERNAME_ENTERING) {
+            "State of the login must be `USERNAME_ENTERING`."
+        }
+        return UsernameEnteringFlow(client, session) { event ->
+            state.value = LoginState.VERIFICATION
+            userCodeReceived = event
+        }
+    }
+
+    /**
+     * Initiates the verification state of the login process.
+     *
+     * @throws IllegalStateException if the state of the login flow is not `VERIFICATION`.
+     */
+    public fun startVerificationFlow(): VerificationFlow {
+        check(state.value == LoginState.VERIFICATION) {
+            "State of the login must be `VERIFICATION`."
+        }
+        return VerificationFlow(client, session, userCodeReceived!!)
+    }
+}
+
+/**
+ * The control flow of the username entering state of the login process.
+ *
+ * @param client enables interaction with the Pingh server.
+ * @param session the information about the current user session.
+ * @param toVerificationState called when user codes are successfully received.
+ */
+public class UsernameEnteringFlow internal constructor(
+    private val client: DesktopClient,
+    private val session: MutableStateFlow<UserSession?>,
+    private val toVerificationState: (UserCodeReceived) -> Unit
+) {
+    /**
+     * Starts the login process and requests `UserCode`.
+     */
+    public fun requestUserCode(
+        username: Username,
+        onSuccess: (event: UserCodeReceived) -> Unit = {}
+    ) {
+        client.requestUserCode(username) { event ->
+            session.value = UserSession(event.id)
+            toVerificationState(event)
+            onSuccess(event)
+        }
+    }
+}
+
+/**
+ * The control flow of the verification state of the login process.
+ *
+ * @param client enables interaction with the Pingh server.
+ * @param session the information about the current user session.
+ * @param event event received in username entering state.
+ */
+@Suppress("MemberVisibilityCanBePrivate") // Some public properties are part
```

Comment:
Please shorten and reformat the suppression according to our discussion in the chat.